harness/determined/common/util.py

```diff
@@ -220,3 +229,23 @@ def wait_for(
         done, rv = predicate()
         time.sleep(interval)
     return rv
+
+
+U = TypeVar("U", bound=Callable[..., Any])
+
+
+def deprecated(message: Optional[str] = None) -> Callable[[U], U]:
```

Comment:
i'm hesitant about introducing a new method to mark deprecation that isn't used by everything else. 

can we just do what is currently done and log a warning? i see that we also have `common.declarative_argparse.warn_on_usage`. if you do decide to keep the decorator, maybe we can combine these two utils that basically do the same thing?

Reply:
I merged the two